.github/actions/python/setup/action.yaml

```
@@ -9,11 +9,13 @@ runs:
   steps:
     - shell: bash
       run: |
-        [[ "${OSTYPE}" =~ "linux" ]] && sudo sed -i 's/azure\.//' /etc/apt/sources.list && sudo apt-get update && sudo apt-get install -y --no-install-recommends libsystemd-dev || echo "do nothing and avoid pipefail"
+        if [[ "${OSTYPE}" =~ "linux" ]]; then
```

This change made because the install of `libsystemd-dev` was failing in our Linux CI builds, which causes the later `pipenv sync` to fail somewhat obscurely when trying to install `systemd-python`.

The failure (GH Actions logs are gone, don't know why) looked a lot like the justification for adding this `sed` in the first place: some Microsoft owned entry in the list was causing `apt` to error out. So, I made two changes:

1. Removed the `sed ... /etc/apt/sources.list` since it didn't appear to be doing anything
2. Moved from an `... && ... || ...` chain to an explicit `if` block so:
    - This step continues to be skipped without prejudice on macOS and Windows
    - This step fails loudly (rather than silently) on Linux builds if something goes wrong with `apt-get install`

I'm pretty sure this was flakiness on the server end (there were 500s and 409s in the logs), so I don't think these changes fixed anything. However, I still thing they're sensible changes